### PR TITLE
Add in_column attribute to all the transforms

### LINKED ISSUE #810: Add `in_column` attribute to all the transforms
### 🚀 Feature Request

Add `in_column` attribute to all the transforms, if there is no such attribute yet. This attribute will be used by the transform to get only the necessary columns from the dataset.

### Proposal

1. Inspect the current transforms and make sure that all of them has attribute `in_column: Union[str, List[str], Literal["all"]]`
2. For transforms from timestamp folder create dummy in_column with default value "target"
3. For `FilterFeaturesTransform` create appropriate in_column base on include/exclude
- If include is not None in_column="all"
- If exclude is not None use in_column=exclude + "target"
5. For feature selection transform create in_column based on features_to_use attribute
6. For segment encoders create in_column with value "target"
7. For nn set in_columns to literal "all"

For the transforms with the constant in_column make in the class attribute, do not initialize it in the constructor

### Test cases

Create test, then check that `in_column` value is generated correctly for transforms from feature_selection folder

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alex-hse-repository**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #810

### CODE REVIEW

File: etna/transforms/feature_selection/filter.py [hunk 1]
```diff
@@ -3,12 +3,15 @@
 
 import pandas as pd
 
+from etna.transforms.base import DymmyInColumnMixin
 from etna.transforms.base import Transform
 
 
-class FilterFeaturesTransform(Transform):
+class FilterFeaturesTransform(Transform, DymmyInColumnMixin):
```
Comment (martins0n) [inline etna/transforms/feature_selection/filter.py:10]: Do we really need DymmyInColumnMixin?

**Review by martins0n — CHANGES_REQUESTED** [review]
* You wanted to make tests for transforms for `feature_selection`?
* May be we dont need this Mixin. It seems we can add dummy `in_column` in base `Transform` for example or add `in_column` attribute separately for all classes. We don't need extra class for it.

File: etna/transforms/feature_selection/filter.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/transforms/feature_selection/filter.py:10]: This mixin indicates the classes with fake or constant in_column, I guess we can leave it now and remove in the future if we do't need it

Comment (alex-hse-repository) [reply etna/transforms/feature_selection/filter.py:10]: I though there would be some additional logic in in_column generation for feature_selection, however there isn't

**Review by martins0n — APPROVED** [review]

**Pull request merged**